### ISSUE TITLE
[BUG] Paperless crashes while starting

### ISSUE THREAD

**Issue by Jefkeny** [astubenbord/paperless-mobile#244]
**Description:**
Easy to explain, the app isn't starting. I just see the logo for hours and nothing happens afterwards.

Tried to click on "back" several times, took a while for it to response, but this just caused a crash of the app.

I've re-installed it already twice, cleared data, checked permissions.
My other phone Samsung Galaxy S7 has connection, but not this one described below.

**Device Information:**
 - Device: Samsung Galaxy S20 FE
 - OS: Android 13 (last update 01.07.23)
 - Paperless Mobile version 2.3.9
 - Paperless Server version 1.16.5

**Comment by astubenbord** [comment]
Any way you could connect your device to a computer, install adb and run `adb logcat | grep flutter`, start the app and send me the logs? That'd help a lot.

**Comment by JigSawFr** [comment]
@astubenbord sometimes I've the same bug on iOS. Suspect a network thing, killing app works after that by I think it can be great to have a screen that showing more infos like "connection to instance..xx in progress".

**Comment by Aliofredi** [comment]
same for me. reinstalled the app and I see only the logo.  no reaction of the app to any attemps to go forward.

(Samsung S23 Ultra / Android)

**Comment by nasana3** [comment]
Same issue here. Reinstall doesn't help. 
(Oneplus 8T / Android 13 Stock Rom)

**Comment by Jefkeny** [comment]
@astubenbord Heyhey, I'm very sorry, but still didn't manage to get you the logs. Maybe one of the others can support?

**Comment by nasana3** [comment]
Hi @astubenbord i paste the log here:

```
08-12 14:02:53.145  9228 11066 I flutter : AuthenticationCubit#restoreSessionState: Trying to restore previous session...
08-12 14:02:53.145  9228 11066 I flutter : AuthenticationCubit#restoreSessionState: Checking if biometric authentication is required...
08-12 14:02:53.145  9228 11066 I flutter : AuthenticationCubit#restoreSessionState: Biometric authentication not configured, skipping.
08-12 14:02:53.145  9228 11066 I flutter : AuthenticationCubit#restoreSessionState: Trying to retrieve authentication credentials...
08-12 14:02:53.152  9228 11066 I flutter : dynamic_color: Core palette detected.
08-12 14:02:53.152  9228 11066 I flutter : Setting refresh rate to 120.00000762939453

08-12 14:02:53.194  9228 11066 I flutter : An unepxected exception has occured!
08-12 14:02:53.194  2286  4830 D DisplayManagerService: performTraversalInternal pendingTraversal
08-12 14:02:53.194  9228 11066 I flutter : PlatformException(Exception encountered, read, javax.crypto.BadPaddingException: error:1e000065:Cipher functions:OPENSSL_internal:BAD_DECRYPT
08-12 14:02:53.194  9228 11066 I flutter :      at com.android.org.conscrypt.NativeCrypto.EVP_CipherFinal_ex(Native Method)
08-12 14:02:53.194  2286  4830 D DisplayDevice: setLayerStackLocked id=local:4630946950956839809 token=android.os.BinderProxy@114b382 stack=0->0
08-12 14:02:53.194  9228 11066 I flutter :      at com.android.org.conscrypt.OpenSSLEvpCipher.doFinalInternal(OpenSSLEvpCipher.java:152)
08-12 14:02:53.194  9228 11066 I flutter :      at com.android.org.conscrypt.OpenSSLCipher.engineDoFinal(OpenSSLCipher.java:374)
08-12 14:02:53.194  9228 11066 I flutter :      at javax.crypto.Cipher.doFinal(Cipher.java:2056)
08-12 14:02:53.194  9228 11066 I flutter :      at e2.h.b(Unknown Source:35)
08-12 14:02:53.194  9228 11066 I flutter :      at d2.a.c(Unknown Source:11)
08-12 14:02:53.194  9228 11066 I flutter :      at d2.a.k(Unknown Source:17)
08-12 14:02:53.194  9228 11066 I flutter :      at d2.e$b.run(Unknown Source:233)
08-12 14:02:53.194  9228 11066 I flutter :      at android.os.Handler.handleCallback(Handler.java:942)
08-12 14:02:53.194  9228 11066 I flutter :      at android.os.Handler.dispatchMessage(Handler.java:99)
08-12 14:02:53.194  9228 11066 I flutter :      at android.os.Looper.loopOnce(Looper.java:240)
08-12 14:02:53.194  2286  2345 D DisplayManagerService: MSG_REQUEST_TRAVERSAL pendingCompleted=false
08-12 14:02:53.194  9228 11066 I flutter :      at android.os.Looper.loop(Looper.java:351)
08-12 14:02:53.194  9228 11066 I flutter :      at android.os.HandlerThread.run(HandlerThread.java:67)
08-12 14:02:53.194  9228 11066 I flutter : , null)
08-12 14:02:53.195  9228 11066 I flutter : #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:652)
08-12 14:02:53.195  9228 11066 I flutter : #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:310)
08-12 14:02:53.195  9228 11066 I flutter : <asynchronous suspension>
08-12 14:02:53.195  9228 11066 I flutter : #2      _getEncryptedBoxKey (package:paperless_mobile/core/config/hive/hive_extensions.dart:34)
08-12 14:02:53.195  9228 11066 I flutter : <asynchronous suspension>
08-12 14:02:53.195  9228 11066 I flutter : #3      withEncryptedBox (package:paperless_mobile/core/config/hive/hive_extensions.dart:14)
08-12 14:02:53.195  9228 11066 I flutter : <asynchronous suspension>
08-12 14:02:53.195  9228 11066 I flutter : #4      AuthenticationCubit.restoreSessionState (package:paperless_mobile/features/login/cubit/authentication_cubit.dart:224)
08-12 14:02:53.195  9228 11066 I flutter : <asynchronous suspension>





08-12 14:05:52.847 10534 11699 I flutter : An unepxected exception has occured!
08-12 14:05:52.847 10534 11699 I flutter : PlatformException(Exception encountered, read, javax.crypto.BadPaddingException: error:1e000065:Cipher functions:OPENSSL_internal:BAD_DECRYPT
08-12 14:05:52.847 10534 11699 I flutter :      at com.android.org.conscrypt.NativeCrypto.EVP_CipherFinal_ex(Native Method)
08-12 14:05:52.847 10534 11699 I flutter :      at com.android.org.conscrypt.OpenSSLEvpCipher.doFinalInternal(OpenSSLEvpCipher.java:152)
08-12 14:05:52.847 10534 11699 I flutter :      at com.android.org.conscrypt.OpenSSLCipher.engineDoFinal(OpenSSLCipher.java:374)
08-12 14:05:52.847 10534 11699 I flutter :      at javax.crypto.Cipher.doFinal(Cipher.java:2056)
08-12 14:05:52.847 10534 11699 I flutter :      at e2.h.b(Unknown Source:35)
08-12 14:05:52.847 10534 11699 I flutter :      at d2.a.c(Unknown Source:11)
08-12 14:05:52.847 10534 11699 I flutter :      at d2.a.k(Unknown Source:17)
08-12 14:05:52.847 10534 11699 I flutter :      at d2.e$b.run(Unknown Source:233)
08-12 14:05:52.847 10534 11699 I flutter :      at android.os.Handler.handleCallback(Handler.java:942)
08-12 14:05:52.847 10534 11699 I flutter :      at android.os.Handler.dispatchMessage(Handler.java:99)
08-12 14:05:52.847 10534 11699 I flutter :      at android.os.Looper.loopOnce(Looper.java:240)
08-12 14:05:52.847 10534 11699 I flutter :      at android.os.Looper.loop(Looper.java:351)
08-12 14:05:52.847 10534 11699 I flutter :      at android.os.HandlerThread.run(HandlerThread.java:67)
08-12 14:05:52.847 10534 11699 I flutter : , null)
08-12 14:05:52.848 10534 11699 I flutter : #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:652)
08-12 14:05:52.848 10534 11699 I flutter : #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:310)
08-12 14:05:52.848 10534 11699 I flutter : <asynchronous suspension>
08-12 14:05:52.848 10534 11699 I flutter : #2      _getEncryptedBoxKey (package:paperless_mobile/core/config/hive/hive_extensions.dart:34)
08-12 14:05:52.848 10534 11699 I flutter : <asynchronous suspension>
08-12 14:05:52.848 10534 11699 I flutter : #3      withEncryptedBox (package:paperless_mobile/core/config/hive/hive_extensions.dart:14)
08-12 14:05:52.848 10534 11699 I flutter : <asynchronous suspension>
08-12 14:05:52.848 10534 11699 I flutter : #4      AuthenticationCubit.restoreSessionState (package:paperless_mobile/features/login/cubit/authentication_cubit.dart:224)
08-12 14:05:52.848 10534 11699 I flutter : <asynchronous suspension>
```

**Comment by astubenbord** [comment]
Hi @nasana3, thanks for the logs! This seems to be an issue with the secure storage plugin (see https://github.com/mogol/flutter_secure_storage/issues/354). I'll implement a workaround once I find the time. But it will probably take another few weeks, sorry guys :/

**Comment by Jefkeny** [comment]
Hi @astubenbord ,

Any suggestions to get it working on android or phone side? Any workaround, not coding i mean, known?

I'd love to use the app again.

**Comment by Enrico204** [comment]
> Hi @astubenbord ,
> 
> Any suggestions to get it working on android or phone side? Any workaround, not coding i mean, known?
> 
> I'd love to use the app again.

I think that uninstalling and reinstalling the app should do the trick. However, if you still have the same problem after that, I think that your problem is not the same as above. In fact, I had the same issue you described due to a network error (the firewall blocked the connection from my phone).

I suggest you to check whether you can reach the Paperless instance from your phone (e.g., with a browser).

If you can attach the phone to a PC and get a trace as described in https://github.com/astubenbord/paperless-mobile/issues/244#issuecomment-1663563735 , that would be very helpful!

@Jefkeny I think that the loading screen should raise some visible error in these cases (a popup?). In a few weeks I will look at the code and I may be able to provide a suggestion :-D

**Comment by Jefkeny** [comment]
Hey @Enrico204,

Re-install didn't change the behaviour and nope no screen, no popup, nothing.

There’s no problem to get access via browser. I can easily open paperless.

Waiting for the next release guys🤞🏻😉🤞🏻😉🤞🏻!

Best regards
Jeff

**Comment by astubenbord** [comment]
Should be fixed in 3.0.1 :)
